k9mail/src/main/java/com/fsck/k9/mailstore/ProvidedTempFileBody.java

```diff
@@ -82,11 +88,11 @@ public long getSize() {
         throw new IllegalStateException("Data must be fully written before it can be read!");
     }
 
-    public File getFile() throws IOException {
+    public Uri getProviderUri(String mimeType) throws IOException {
```

Comment:
No, AttachmentInfoExtractor knows how to work with a ProvidedTempFileBody, which is a body it can get a provider URI from. It knows nothing about the particular FileProvider the body it backed by - and neither does the body itself.